### PR TITLE
PMM-7697 Backup flag

### PR DESCRIPTION
**Pull request by Dasio**
[PMM-7697](https://jira.percona.com/browse/PMM-7697)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1629

- [ ] Links to other linked pull requests (optional).
- API: https://github.com/percona/pmm/pull/711
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Comment by Dasio** [comment]
Failed tests (TestPlatformService) are not related to this PR

File: models/settings_helpers.go [hunk 1]
```diff
@@ -257,6 +264,14 @@ func UpdateSettings(q reform.DBTX, params *ChangeSettingsParams) (*Settings, err
 		settings.IntegratedAlerting.SlackAlertingSettings = params.SlackAlertingSettings
 	}
 
+	if params.DisableBackupManagement {
```
Comment (oter) [inline models/settings_helpers.go:267]: is there a check that these two fields cannot be true at the same time?

**Review by oter — APPROVED** [review]

**Review by oter — COMMENTED** [review]
Let's fix PR title "PMM-XXXX: Title"

File: models/settings_helpers.go (hunk 1 above)
Comment (Dasio) [reply models/settings_helpers.go:267]: No, can't see this check anywhere. Same for DBaaS,VMCache, Alerting..

Comment (artemgavrilov) [reply models/settings_helpers.go:267]: Here it is)
https://github.com/percona/pmm-managed/blob/461cb41817f9838d7fde755ce799e1211eb22ac5/models/settings_helpers.go#L269-L281

Comment (Dasio) [reply models/settings_helpers.go:267]: :facepalm: I was looking on to it before... thx

**Review by artemgavrilov — APPROVED** [review]

File: services/management/backup/artifacts_service.go [hunk 2]
```diff
@@ -19,6 +19,8 @@ package backup
 import (
 	"context"
 
+	"github.com/sirupsen/logrus"
+
```
Comment (artemgavrilov) [inline services/management/backup/artifacts_service.go:23]: Remove newline

File: services/management/backup/locations_service.go [hunk 3]
```diff
@@ -19,6 +19,8 @@ package backup
 import (
 	"context"
 
+	"github.com/sirupsen/logrus"
+
```
Comment (artemgavrilov) [inline services/management/backup/locations_service.go:23]: Remove new line

Comment (Dasio) [reply services/management/backup/locations_service.go]: Interesting that goimports doesn't handle it

Comment (artemgavrilov) [reply services/management/backup/locations_service.go]: Goimports allows `custom` groups, for some reason ¯\_(ツ)_/¯

File: utils/interceptors/service.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+			}
+		}
+		return handler(ctx, req)
+	}
+}
+
+// StreamServiceEnabledInterceptor returns a new unary server interceptor that checks if service is enabled.
+func StreamServiceEnabledInterceptor() grpc.StreamServerInterceptor {
+	return func(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+		if svc, ok := srv.(serviceEnabled); ok {
```
Comment (BupycHuk) [inline utils/interceptors/service.go:33]: can we just simplify it like?
```suggestion
		if svc, ok := srv.(serviceEnabled); ok && !svc.Enabled() {
```

**Review by BupycHuk — APPROVED** [review]

**Review by BupycHuk — CHANGES_REQUESTED** [review]
Please update license headers.

File: services/management/backup/artifacts_service.go [hunk 5]
```diff
@@ -23,23 +23,36 @@ import (
...
+		l:  logrus.WithField("component", "management/backup/artifacts"),
 		db: db,
 	}
 }
 
+// Enabled returns if service is enabled and can be used.
+func (s *ArtifactsService) Enabled() bool {
+	settings, err := models.GetSettings(s.db)
+	if err != nil {
+		s.l.WithError(err).Error("enabled: get settings")
```
[the remark this answers is not in the document]
Comment (Dasio) [reply services/management/backup/artifacts_service.go:50]: Any idea?

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**